php-checks/src/test/resources/checks/security/DataEncryptionCheck.php

```diff
@@ -208,3 +208,6 @@ function myLaravelEncrypt($data)
     encrypt($data); // Noncompliant
     ABC::encryptString($data); // Ok - not from framework
 }
+
+#[MyAttribute(null, new stdClass)]
+class TestClassWithAttribute {} // SONARPHP-1153
```

Comment:
I believe other cases cause a failure:
```
$x = new #[A(new X1)] class {}; // anonymous class
function foo(#[A(new X2)] $param) {} // parameter
```
I would check all tree classes which have instances of `AttributeGroupTree` and for which `SymbolVisitor` overrides the `visit*` method without calling `super.visit*`.

I wonder if there would be a generic way to test that we don't miss any `NamespaceNameTree`. Something similar to `PHPTreeModelTest.checkFullFidelity` which ensures that our AST doesn't miss any token for all parsing tests which use `PHPTreeModelTest.parse`. Could we improve `PHPTreeModelTest.parse` to do that?

Reply:
Thanks for noticing the two missing places! They are considered now too (ed13443).

> I wonder if there would be a generic way to test that we don't miss any NamespaceNameTree. Something similar to PHPTreeModelTest.checkFullFidelity which ensures that our AST doesn't miss any token for all parsing tests which use PHPTreeModelTest.parse. Could we improve PHPTreeModelTest.parse to do that?

I see that this could be useful. I believe we had some cases in the past where we missed symbols on NamespaceNameTree (e.g., https://jira.sonarsource.com/browse/SONARPHP-1082). 
I'm not sure however to fully understand the possible approach with `PHPTreeModelTest.parse` you have in mind. We do not create the symbol table in that method, and it would be difficult to do it because currently our `SymbolTableImpl.create` works only on `CompilationUnitTree`, but in `PHPTreeModelTest.parse` we mostly create different tree kinds. If I totally misunderstood the idea, happy to talk about it :smile: